### DISCUSSION TITLE
在微信开发者工具中运行报错“module  xxx is not defined”

### DISCUSSION THREAD

**Discussion by OpenWJY** [NervJS/taro#14595] [Q & A]
## 版本一览
+ Taro：v3.6.17
+ Taro模板：默认模板
+ 微信小程序开发者工具：1.06.2308310win32-x64
+ 微信小程序调试基础库：3.1.0

## 问题概述
按照官方[开始文档](https://docs.taro.zone/docs/GETTING-STARTED#%E9%A1%B9%E7%9B%AE%E5%88%9D%E5%A7%8B%E5%8C%96)构建了项目，并编译运行，使用微信开发者工具进行预览时报错：
```
VM1361:9 app.js错误:
 Error: module 'vendors-node_modules_taro_weapp_prebundle_react-dom_js.js' is not defined, require args is './vendors-node_modules_taro_weapp_prebundle_react-dom_js.js'
    at V (WASubContext.js?t=wechat&s=1695608512793&v=3.0.2:1)
    at n (WASubContext.js?t=wechat&s=1695608512793&v=3.0.2:1)
    at app.js:3
    at WASubContext.js?t=wechat&s=1695608512793&v=3.0.2:1
    at d.runWith (WASubContext.js?t=wechat&s=1695608512793&v=3.0.2:1)
    at V (WASubContext.js?t=wechat&s=1695608512793&v=3.0.2:1)
    at <anonymous>:7:9
    at doWhenAllScriptLoaded (<anonymous>:6261:21)
    at Object.scriptLoaded (<anonymous>:6289:5)
    at Object.<anonymous> (<anonymous>:6314:22)(env: Windows,mp,1.06.2308310; lib: 3.0.2)
```

微信开发者工具报错图片：
![image](https://github.com/NervJS/taro/assets/29328336/94e2af48-2c72-4f07-b8de-a83a82dfd8de)

项目目录图片：
![image](https://github.com/NervJS/taro/assets/29328336/308906a7-2cf4-428b-a80c-f7db4450ec05)

## 自查补充
1. 运行的历史命令有：
  ```bash
  taro init myApp
  
  taro build --type weapp --watch
  ```
2. 报错信息中已经表明是 `app.js` 中的 js 引用未找到，翻看 `dist` 目录，目标 js 实际是在 `prebundle` 目录下，手动修改后，小程序显示出了默认模板的 Hello world 字样。然而，`dist` 目录是脚手架自己构建出来的，还需二次修改显然不太丝滑。
  ```javascript
  # 例子：修改前 
  require("./vendors-node_modules_taro_weapp_prebundle_react-dom_js.js");
  
  # 修改后
  require("./prebundle/vendors-node_modules_taro_weapp_prebundle_react-dom_js.js");
  ```

3. 微信开发者工具中的几个官方提到的勾选项都已关闭

## 期望得到的帮助
目前是按照官方文档一步步执行下来的，在文档常见问题和相关论坛讨论中暂未搜到期望的解决。
是缺少几项关键的编译配置吗？如何配置能解决构建出的 dist/app.js 找不到引用的问题？

**Comment by OpenWJY** [comment]
## 记录
根据朋友帮助，目前的临时处理方式为关闭预编译，修改 `./config/index.ts` 中 `compiler` 相关配置。
```typescript
# 原来
compiler: 'webpack5'

# 修改后
compiler: {
  type: 'webpack5',
  prebundle: {
    enable: false,
  }
},
```

**Reply by yudinosaur** [reply]
但是这样的话，每次都得点击重新编译，页面才会刷新，不会热更新了

**Comment by Typeve** [comment]
这个预编译选项从来没有成功开启过
现在开发项目编译时间太长了

**Comment by 773579084** [comment]
我也是遇到这个问题，taro问题真的多，体验没有uniapp好，可惜uniapp不支持react

**Reply by OpenWJY** [reply]
用爱发电的项目，需要包容他的不足

**Reply by 773579084** [reply]
确实要多包容，就是太痛苦了😂

**Comment by lxhyz** [comment]
这个问题在新版貌似还是没有解决
